app/models/repo_solr_client.rb

```diff
@@ -83,4 +83,54 @@ def update_index_and_delete_empty_parents(new_document)
     @rsolr.add new_document
     @rsolr.commit
   end
+  
+  # remove all captures not updated in current run to ensure bad captures are deleted -- use wisely!
+  def delete_unseen_captures_below(item_uuid, seen_uuids)
+    return unless @rsolr
+    
+    query = 'type_s:Capture AND immediateParent_s:"' + item_uuid + '"'
+
+    # Fetch the initial response to determine the total number of results
+    resp = @rsolr.get('select', params: { q: query, rows: 0 })
+    
+    unless resp['response']
+      raise "Bad response from Solr for immediateParent_s:#{item_uuid}."
+    end
+    
+    total_results = resp['response']['numFound']
+    deletes = false
+
+    # Calculate the total pages to loop through
+    total_pages = (total_results + 249) / 250
+
+    # Initialize variables for pagination
+    page = 0
+
+    while page <= total_pages do
+      # Set the start parameter for pagination
+      start = page * 250
+  
+      # Fetch documents from Solr with pagination
+      response = @rsolr.get('select', params: { q: query, start: start, rows: 250 })
+      
+      unless response['response']
+        puts "Bad response from Solr for page #{page}. Skipping."
```

Comment:
Is skipping a page the desired behavior with a bad response, or would it make sense to raise an exception?

Reply:
Would make more sense to raise an exception, yep.